src/main/java/org/opensearch/ad/feature/SearchFeatureDao.java

```diff
@@ -158,6 +161,62 @@ public SearchFeatureDao(
         );
     }
 
+    public SearchFeatureDao(
+        Client client,
+        NamedXContentRegistry xContent,
+        Interpolator interpolator,
+        ClientUtil clientUtil,
+        Settings settings,
+        ClusterService clusterService,
+        int minimumDocCount,
+        ExtensionRunner extensionRunner
+    ) {
+        this(
+            client,
+            xContent,
+            interpolator,
+            clientUtil,
+            settings,
+            clusterService,
+            minimumDocCount,
+            Clock.systemUTC(),
+            MAX_ENTITIES_FOR_PREVIEW.get(settings),
+            PAGE_SIZE.get(settings),
+            PREVIEW_TIMEOUT_IN_MILLIS,
+            extensionRunner
+        );
+    }
+
+    public SearchFeatureDao(
+        Client client,
+        TransportService transportService,
+        Interpolator interpolator,
+        ClientUtil clientUtil,
+        Settings settings,
+        ClusterService clusterService,
+        int minimumDocCount,
+        Clock clock,
+        int maxEntitiesForPreview,
+        int pageSize,
+        long previewTimeoutInMilliseconds,
+        ExtensionRunner extensionRunner
+    ) {
+        this.client = client;
+        this.interpolator = interpolator;
+        this.clientUtil = clientUtil;
+        this.maxEntitiesForPreview = maxEntitiesForPreview; 
+        this.pageSize = pageSize;
+        Map<Setting<?>, Consumer<?>> settingUpdateConsumers = new HashMap<Setting<?>, Consumer<?>>();
+        Consumer<Integer> maxEntitiesForPreviewConsumer = it -> this.maxEntitiesForPreview = it;
+        Consumer<Integer> pageSizeConsumer = it -> this.pageSize = it;
+
+        extensionRunner.sendAddSettingsUpdateConsumerRequest(transportService, settingUpdateConsumers);
```

Comment:
Lets add the consumers and settings to the `settingUpdateConsumers` map prior to invoking `sendAddSettingsUpdateConsumerRequest`

Reply:
That might be important.